### PR TITLE
Refactor hash state system

### LINKED ISSUE #1134: First-time About doesn't close (transient)
I'm occasionally getting a first-time cookie expire that's causing About to display.

Seemingly, when this happens, I can't close About without lots of twiddling (like alt-tab swapping tabs or window, editing URL, opening JSconsole).. in other words, not sure what's causing it to freeze like that

### PR DESCRIPTION
**Pull request by pablo-mayrgundter**
- Fixes #1134, which had an e2e test already at view-100/initial-model-load-and-view.cy.js `AboutDialog not visible, model visible - Screen`, so the bug is indeed transient and I don't think I can add any new test for it
- Each component with nav state (e.g. NavTree, PropertiesPanel, Notes, Share dialog, About dialog, etc):
  - has its own directory (some already did)
  - has a consistent definition of the `HASH_PREFIX_<component name>` variable in a `hashState.js` file in its directory
  - if the component state can be passed in a URL, then has isVisible zustand var set to isVisibleInitially, which is defined in each component's hashState.js file.  Usually this true if the hash state is present, but can also be true from a cookie, ie for the About state for visitors without isFirstTime set.  That's what fixes #1134.
- Change CutPlane HASH_PREFIX from `vp` to `cp`.  Maybe should be `cut`?
- In location.js add passthru helpers getParams, hasParams, setParams, etc that passthru to getHashParams, hasHashParams, setHashParams, but without passing location
- Fixup cy-percy script to accept spec
- Disable flaky test and created #1178

# Visual diff e2e tests
Full run: https://percy.io/8fe2b2f1/share/builds/34111775/changed/1867319070

After fixing 2 regressions, just ran those alone: https://percy.io/8fe2b2f1/share/builds/34111978/changed/1867332591

```
yarn install && yarn cy-build && yarn cy-percy cypress/e2e/open/100/open-model-from-gh-ui.cy.js cypress/e2e/view-100/cutplanes.cy.js
```

### CODE REVIEW

**Review by OlegMoshkovich — CHANGES_REQUESTED** [review]
A lot of work and nice improvements!
When testing the behavior I found the following issues:

Test System:
Version 124.0.6367.119 (Official Build) (x86_64)

## 1. Search Hash Functionality
_Issue_: The Search bar does not display when the link containing the search hash is opened in the new session.
## 2. Notes Hash Functionality
_Issue_: When a specific note is selected and a new model is loaded into the session the notes do not navigate to the notes list

**Steps to Reproduce:**
- Select a note on the home screen.
- Switch to Momentum
Observed Behavior: Side - drawer displays an empty drawer.
Expected Behavior:  Side - drawer should update to show a list of notes from the newly loaded model.
## 4. Share Dialog Functionality
_Issue_: The share dialog appears when a share link is obtained through the share dialog. 
When the link is opened, instead of accessing the shared content directly, users are presented with the share dialog. This can be confusing and disrupts the user experience.
## 5. Imagine Dialog Issue
_Issue_: When opening a link containing the "imagine" hash, the dialog is displayed but remains empty with no screenshot of the current model view visible

**Steps to Reproduce:**
- Open the imagine dialog
- Copy the URL
- Paste the URL in the new session
Observed Behavior: Imagine dialog is displayed with a blank screenshotExpected Behavior:  Imagine dialog is displayed with a screenshot of the current model view

File: src/Components/Buttons.jsx [hunk 1]
```diff
@@ -141,15 +155,17 @@ export function ControlButtonWithHashState({
   // Enforce invariant
   useEffect(() => {
     if (isDialogDisplayed) {
+      console.log('Buttons#ControlButtonWithHashState#useEffect, isDialogDisplayed: true, adding hash param')
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/Components/Buttons.jsx:158]: Done

**Comment by pablo-mayrgundter** [comment]
Addressed 1-Search, 2-Notes, 4-Imagine.. 3 is existing behavior and I tend to agree but that'll take a bit of work so suggest we track separately

File: cypress/e2e/search/100/permalink.cy.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import '@percy/cypress'
+import {waitForModelReady} from '../../../support/models'
+import {
+  homepageSetup,
+  setIsReturningUser,
+  visitHomepageWaitForModel,
+} from '../../../support/utils'
+
+
+/** {@link } */
```
Comment (pablo-mayrgundter) [inline cypress/e2e/search/100/permalink.cy.js:10]: pending

**Comment by OlegMoshkovich** [comment]
Love the search and imagine improvements!
It is all working but the notes issues, it still do not reset when a note is selected and a new model is loaded.

**Review by OlegMoshkovich — APPROVED** [review]

**Comment by OlegMoshkovich** [comment]
Approved.
This is still happening, even though the hash is reset when a new model is loaded.
2 Notes Hash
Issue: When a specific note is selected and a new model is loaded into the session the notes do not navigate to the notes list

Steps to Reproduce:

Select a note on the home screen.
Switch to Momentum
Observed Behavior: The side - drawer displays an empty drawer.
Expected Behavior: The side - drawer should update to show a list of notes from the newly loaded model.

image

**Pull request merged**